test/test_metadata.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+import unittest
+
+from picard import config
+from picard.metadata import Metadata, MULTI_VALUED_JOINER
+
+
+settings = {
+    'write_id3v23': False,
+    'id3v23_join_with': '/',
+}
+
+
+class MetadataTest(unittest.TestCase):
+
+    original = None
+    tags = []
+
+    def setUp(self):
+        config.setting = settings.copy()
+        self.metadata = Metadata()
+        self.metadata["single1"] = "single1-value"
+        self.metadata.add_unique("single2", "single2-value")
+        self.metadata.add_unique("single2", "single2-value")
+        self.multi1 = ["multi1-value", "multi1-value"]
+        self.metadata.add("multi1", self.multi1[0])
+        self.metadata.add("multi1", self.multi1[1])
+        self.multi2 = ["multi2-value1", "multi2-value2"]
+        self.metadata["multi2"] = self.multi2
+        self.multi3 = ["multi3-value1", "multi3-value2"]
+        self.metadata.set("multi3", self.multi3)
+        self.metadata["~hidden"] = "hidden-value"
+
+    def tearDown(self):
+        pass
+
+    def test_metadata_set(self):
+        self.assertEqual(["single1-value"], dict.get(self.metadata,"single1"))
+        self.assertEqual(["single2-value"], dict.get(self.metadata,"single2"))
+        self.assertEqual(self.multi1, dict.get(self.metadata,"multi1"))
+        self.assertEqual(self.multi2, dict.get(self.metadata,"multi2"))
+        self.assertEqual(self.multi3, dict.get(self.metadata,"multi3"))
+        self.assertEqual(["hidden-value"], dict.get(self.metadata,"~hidden"))
+
+    def test_metadata_get(self):
+        self.assertEqual("single1-value", self.metadata["single1"])
+        self.assertEqual("single1-value", self.metadata.get("single1"))
+        self.assertEqual(["single1-value"], self.metadata.getall("single1"))
+
+        self.assertEqual(MULTI_VALUED_JOINER.join(self.multi1), self.metadata["multi1"])
+        self.assertEqual(MULTI_VALUED_JOINER.join(self.multi1), self.metadata.get("multi1"))
+        self.assertEqual(self.multi1, self.metadata.getall("multi1"))
+
+        self.assertEqual("", self.metadata["nonexistent"])
+        self.assertEqual(None, self.metadata.get("nonexistent"))
+        self.assertEqual([], self.metadata.getall("nonexistent"))
+
+        self.assertEqual(dict.items(self.metadata), self.metadata.rawitems())
+        metadata_items = [(x, z) for (x, y) in dict.items(self.metadata) for z in y]
```

Comment:
Yes - this is to create (using dict) the data that metadata.items is tested against. If I used metadata items to create it then it wouldn't be a valid test.